### PR TITLE
LibRetro.RetroArch.Nightly.64-bit

### PR DESCRIPTION
**Pull request by fpscan**
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`? 
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.1 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.1.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----
image

I have this [PR](https://github.com/microsoft/winget-pkgs/pull/59175) open but I decided to go with separate architecture PR for future proof. While that PR open but fails installer HASH validation, this PR uses same links and everything but doesn't fail installer hash on my local environment. So I'm fine with either one of these PR get merged so I can ninja release on our Stable so I can make PR for stable version.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/59312)

### CODE REVIEW

**Comment by fpscan** [comment]
Hi @ItzLevvie this could be just me but I wasn't expected this is going to be that much exhausting. Could you review this one? First [PR](https://github.com/microsoft/winget-pkgs/pull/59175) contains both 32bit and 64bit builds but fails [hash validation](https://dev.azure.com/ms/winget-pkgs/_build/results?buildId=307382&view=logs&j=871b4eed-22de-5cb6-c1c8-82f190de633f&t=523ceef4-ddf7-55a0-c83d-dd0f77b67744&l=65). This PR is using same build but only for 64bit, however fails at [installation](https://dev.azure.com/ms/winget-pkgs/_build/results?buildId=307388&view=logs&j=871b4eed-22de-5cb6-c1c8-82f190de633f&t=58bdf892-4e45-5ba7-4782-31912f95e167&l=38). Anychance manual approve can be considered at this point? Thanks in advance for your review.

**Comment by Trenly** [comment]
It is still failing the SHA256 validation, but just in a different place

```
2022-05-01 01:41:48.007 [CLI ] WinGet invoked with arguments: 'install' '--manifest' 'D:\TRANSFER\ab04b65d-2ac9-ec11-950a-000d3a9ed19d\Manifest.yaml' '--accept-package-agreements'
2022-05-01 01:41:48.008 [CLI ] Found subcommand: install
2022-05-01 01:41:48.008 [CLI ] Leaf command to execute: root:install
2022-05-01 01:41:48.021 [CLI ] Executing command: install
2022-05-01 01:41:48.024 [CLI ] Manifest fields: Name [RetroArch Nightly], Version [1.10.3.0]
2022-05-01 01:41:48.028 [CLI ] Starting installer selection.
2022-05-01 01:41:48.028 [CLI ] Completed installer selection.
2022-05-01 01:41:48.067 [CLI ] Generated temp download path: C:\Users\VALIDA~1\AppData\Local\Temp\WinGet\Libretro.RetroArch.Nightly.64-bit.1.10.3.0\aab8a8bd4a382011c9417dbc363db98cf2be49ab8d0eff103b93978fe805c64a
2022-05-01 01:41:48.068 [CORE] Downloading to path: C:\Users\VALIDA~1\AppData\Local\Temp\WinGet\Libretro.RetroArch.Nightly.64-bit.1.10.3.0\aab8a8bd4a382011c9417dbc363db98cf2be49ab8d0eff103b93978fe805c64a
2022-05-01 01:41:48.068 [CORE] DeliveryOptimization downloading from url: https://buildbot.libretro.com/nightly/windows/x86_64/RetroArch-Win64-setup.exe
2022-05-01 01:41:50.695 [CORE] Download completed.
2022-05-01 01:42:04.504 [CORE] Started applying motw to C:\Users\VALIDA~1\AppData\Local\Temp\WinGet\Libretro.RetroArch.Nightly.64-bit.1.10.3.0\aab8a8bd4a382011c9417dbc363db98cf2be49ab8d0eff103b93978fe805c64a with zone: 3
2022-05-01 01:42:04.515 [CORE] Finished applying motw
2022-05-01 01:42:04.515 [CLI ] Package hash verification failed. SHA256 in manifest [aab8a8bd4a382011c9417dbc363db98cf2be49ab8d0eff103b93978fe805c64a] does not match download [be29572a70d4f90af16c8990d6235fa7c375fa8de1aaa21ecfd3961acb5e2d88]
```

**Comment by fpscan** [comment]
> It is still failing the SHA256 validation, but just in a different place
> 
> ```
> 2022-05-01 01:41:48.007 [CLI ] WinGet invoked with arguments: 'install' '--manifest' 'D:\TRANSFER\ab04b65d-2ac9-ec11-950a-000d3a9ed19d\Manifest.yaml' '--accept-package-agreements'
> 2022-05-01 01:41:48.008 [CLI ] Found subcommand: install
> 2022-05-01 01:41:48.008 [CLI ] Leaf command to execute: root:install
> 2022-05-01 01:41:48.021 [CLI ] Executing command: install
> 2022-05-01 01:41:48.024 [CLI ] Manifest fields: Name [RetroArch Nightly], Version [1.10.3.0]
> 2022-05-01 01:41:48.028 [CLI ] Starting installer selection.
> 2022-05-01 01:41:48.028 [CLI ] Completed installer selection.
> 2022-05-01 01:41:48.067 [CLI ] Generated temp download path: C:\Users\VALIDA~1\AppData\Local\Temp\WinGet\Libretro.RetroArch.Nightly.64-bit.1.10.3.0\aab8a8bd4a382011c9417dbc363db98cf2be49ab8d0eff103b93978fe805c64a
> 2022-05-01 01:41:48.068 [CORE] Downloading to path: C:\Users\VALIDA~1\AppData\Local\Temp\WinGet\Libretro.RetroArch.Nightly.64-bit.1.10.3.0\aab8a8bd4a382011c9417dbc363db98cf2be49ab8d0eff103b93978fe805c64a
> 2022-05-01 01:41:48.068 [CORE] DeliveryOptimization downloading from url: https://buildbot.libretro.com/nightly/windows/x86_64/RetroArch-Win64-setup.exe
> 2022-05-01 01:41:50.695 [CORE] Download completed.
> 2022-05-01 01:42:04.504 [CORE] Started applying motw to C:\Users\VALIDA~1\AppData\Local\Temp\WinGet\Libretro.RetroArch.Nightly.64-bit.1.10.3.0\aab8a8bd4a382011c9417dbc363db98cf2be49ab8d0eff103b93978fe805c64a with zone: 3
> 2022-05-01 01:42:04.515 [CORE] Finished applying motw
> 2022-05-01 01:42:04.515 [CLI ] Package hash verification failed. SHA256 in manifest [aab8a8bd4a382011c9417dbc363db98cf2be49ab8d0eff103b93978fe805c64a] does not match download [be29572a70d4f90af16c8990d6235fa7c375fa8de1aaa21ecfd3961acb5e2d88]
> ```

How can I fix this? Should I just replace current SHA with `be29572a70d4f90af16c8990d6235fa7c375fa8de1aaa21ecfd3961acb5e2d88`

**Comment by Trenly** [comment]
I’m not sure, because when I download the file I get a hash of `494bbf97e55299f79f89e61a350cf31a76f07c22cb71b4c2cc913e5a57ae2321`

Is there a dynamic CDN which changes the file that gets downloaded? Could it be a CDN caching issue?

**Comment by fpscan** [comment]
> I’m not sure, because when I download the file I get a hash of `494bbf97e55299f79f89e61a350cf31a76f07c22cb71b4c2cc913e5a57ae2321`
> 
> Is there a dynamic CDN which changes the file that gets downloaded? Could it be a CDN caching issue?

I also get `494BBF97E55299F79F89E61A350CF31A76F07C22CB71B4C2CC913E5A57AE2321`. I'll check CDN later today. I believe there is nothing else we can do at this moment(?)

**Comment by j8srv32** [comment]
Yes

**Comment by j8srv32** [comment]
With the SHA256 sum, you need to change it to the sum that winget got. What CDN are you using also?

File: manifests/l/Libretro/RetroArch/Nightly/64-bit/1.10.3.0/Libretro.RetroArch.Nightly.64-bit.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Created using wingetcreate 1.0.4.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.1.0.schema.json
+
+PackageIdentifier: Libretro.RetroArch.Nightly.64-bit
+PackageVersion: 1.10.3.0
+Installers:
+- Architecture: x64
+  InstallerType: nullsoft
+  InstallerUrl: https://buildbot.libretro.com/nightly/windows/x86_64/RetroArch-Win64-setup.exe
```
Comment (j8srv32) [inline manifests/l/Libretro/RetroArch/Nightly/64-bit/1.10.3.0/Libretro.RetroArch.Nightly.64-bit.installer.yaml:9]: Not the right URL for the version 1.10.3, URL is https://buildbot.libretro.com/stable/1.10.3/windows/x86_64/RetroArch-Win64-setup.exe

File: manifests/l/Libretro/RetroArch/Nightly/64-bit/1.10.3.0/Libretro.RetroArch.Nightly.64-bit.installer.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# Created using wingetcreate 1.0.4.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.1.0.schema.json
+
+PackageIdentifier: Libretro.RetroArch.Nightly.64-bit
+PackageVersion: 1.10.3.0
```
Comment (j8srv32) [inline manifests/l/Libretro/RetroArch/Nightly/64-bit/1.10.3.0/Libretro.RetroArch.Nightly.64-bit.installer.yaml:5]: Wrong version. Should be 1.10.3

Comment (j8srv32) [reply manifests/l/Libretro/RetroArch/Nightly/64-bit/1.10.3.0/Libretro.RetroArch.Nightly.64-bit.installer.yaml:5]: Also, SHA256 sums are: 7fae3676a37a2b174692c9fc3d17b0c4db284966049e8630cf93a004aa588c60

Comment (j8srv32) [reply manifests/l/Libretro/RetroArch/Nightly/64-bit/1.10.3.0/Libretro.RetroArch.Nightly.64-bit.installer.yaml:5]: Not the right URL for the version 1.10.3, URL is https://buildbot.libretro.com/stable/1.10.3/windows/x86_64/RetroArch-Win64-setup.exe

File: manifests/l/Libretro/RetroArch/Nightly/64-bit/1.10.3.0/Libretro.RetroArch.Nightly.64-bit.locale.en-US.yaml [hunk 3]
```diff
@@ -0,0 +1,26 @@
+# Created using wingetcreate 1.0.4.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.defaultLocale.1.1.0.schema.json
+
+PackageIdentifier: Libretro.RetroArch.Nightly.64-bit
+PackageVersion: 1.10.3.0
```
Comment (j8srv32) [inline manifests/l/Libretro/RetroArch/Nightly/64-bit/1.10.3.0/Libretro.RetroArch.Nightly.64-bit.locale.en-US.yaml:5]: Version should be 1.10.3

File: manifests/l/Libretro/RetroArch/Nightly/64-bit/1.10.3.0/Libretro.RetroArch.Nightly.64-bit.yaml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+# Created using wingetcreate 1.0.4.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.version.1.1.0.schema.json
+
+PackageIdentifier: Libretro.RetroArch.Nightly.64-bit
+PackageVersion: 1.10.3.0
```
Comment (j8srv32) [inline manifests/l/Libretro/RetroArch/Nightly/64-bit/1.10.3.0/Libretro.RetroArch.Nightly.64-bit.yaml:5]: Version should be 1.10.3

**Review by j8srv32 — CHANGES_REQUESTED** [review]
This is an invalid review. Use the next one (it does not have bugs) and, this has bugs.

File: manifests/l/Libretro/RetroArch/Nightly/64-bit/1.10.3.0/Libretro.RetroArch.Nightly.64-bit.installer.yaml [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+Installers:
+- Architecture: x64
+  InstallerType: nullsoft
+  InstallerUrl: https://buildbot.libretro.com/nightly/windows/x86_64/RetroArch-Win64-setup.exe
+  InstallerSha256: AAB8A8BD4A382011C9417DBC363DB98CF2BE49AB8D0EFF103B93978FE805C64A
```
Comment (j8srv32) [inline manifests/l/Libretro/RetroArch/Nightly/64-bit/1.10.3.0/Libretro.RetroArch.Nightly.64-bit.installer.yaml:10]: Wrong SHA256 sum. Needs 7fae3676a37a2b174692c9fc3d17b0c4db284966049e8630cf93a004aa588c60

File: manifests/l/Libretro/RetroArch/Nightly/64-bit/1.10.3.0/Libretro.RetroArch.Nightly.64-bit.installer.yaml (hunk 1 above)
Comment (j8srv32) [inline manifests/l/Libretro/RetroArch/Nightly/64-bit/1.10.3.0/Libretro.RetroArch.Nightly.64-bit.installer.yaml:9]: Wrong URL. Should be https://buildbot.libretro.com/stable/1.10.3/windows/x86_64/RetroArch-Win64-setup.exe

**Review by j8srv32 — CHANGES_REQUESTED** [review]
invalid

File: manifests/l/Libretro/RetroArch/Nightly/64-bit/1.10.3.0/Libretro.RetroArch.Nightly.64-bit.installer.yaml (hunk 5 above)
Comment (j8srv32) [inline manifests/l/Libretro/RetroArch/Nightly/64-bit/1.10.3.0/Libretro.RetroArch.Nightly.64-bit.installer.yaml:10]: Wrong SHA256 sum. You need 7fae3676a37a2b174692c9fc3d17b0c4db284966049e8630cf93a004aa588c60 for it to work.

**Review by j8srv32 — CHANGES_REQUESTED** [review]
Doing final draft.

File: manifests/l/Libretro/RetroArch/Nightly/64-bit/1.10.3.0/Libretro.RetroArch.Nightly.64-bit.installer.yaml (hunk 5 above)
Comment (j8srv32) [inline manifests/l/Libretro/RetroArch/Nightly/64-bit/1.10.3.0/Libretro.RetroArch.Nightly.64-bit.installer.yaml:10]: Wrong SHA256 sum. You need 7fae3676a37a2b174692c9fc3d17b0c4db284966049e8630cf93a004aa588c60 for it to work.
Line 10...

File: manifests/l/Libretro/RetroArch/Nightly/64-bit/1.10.3.0/Libretro.RetroArch.Nightly.64-bit.installer.yaml (hunk 1 above)
Comment (j8srv32) [inline manifests/l/Libretro/RetroArch/Nightly/64-bit/1.10.3.0/Libretro.RetroArch.Nightly.64-bit.installer.yaml:9]: Wrong URL. Should be https://buildbot.libretro.com/stable/1.10.3/windows/x86_64/RetroArch-Win64-setup.exe
Line 9...

File: manifests/l/Libretro/RetroArch/Nightly/64-bit/1.10.3.0/Libretro.RetroArch.Nightly.64-bit.yaml (hunk 4 above)
Comment (j8srv32) [inline manifests/l/Libretro/RetroArch/Nightly/64-bit/1.10.3.0/Libretro.RetroArch.Nightly.64-bit.yaml:5]: Wrong version, should be 1.10.3 
Line 5...

File: manifests/l/Libretro/RetroArch/Nightly/64-bit/1.10.3.0/Libretro.RetroArch.Nightly.64-bit.locale.en-US.yaml (hunk 3 above)
Comment (j8srv32) [inline manifests/l/Libretro/RetroArch/Nightly/64-bit/1.10.3.0/Libretro.RetroArch.Nightly.64-bit.locale.en-US.yaml:5]: Wrong version, should be 1.10.3 
Line 5...

**Review by j8srv32 — CHANGES_REQUESTED** [review]
almost done

File: manifests/l/Libretro/RetroArch/Nightly/64-bit/1.10.3.0/Libretro.RetroArch.Nightly.64-bit.installer.yaml (hunk 2 above)
Comment (j8srv32) [inline manifests/l/Libretro/RetroArch/Nightly/64-bit/1.10.3.0/Libretro.RetroArch.Nightly.64-bit.installer.yaml:5]: Wrong version, should be 1.10.3
Line 5...

**Review by j8srv32 — COMMENTED** [review]
Ok, submitting final draft.

**Review by j8srv32 — CHANGES_REQUESTED** [review]
# Fix bugs found

- [ ] Have you added the new Uniform Resource Locator (URL)?
- [ ] Have you added the new SHA-256 sum?
- [ ] Have you added the new version?

# How to re-run this
Type, in a new comment, "/AzurePipelines run".
### Goodbye! Have a good day! :)

**Comment by j8srv32** [comment]
@msftbot I have reviewed the changes. @fpscan needs to fix these for the pull request (PR) to succeed.

**Comment by j8srv32** [comment]
The problem is, you cannot use Nightly builds, you would have to change the hash every day...

**Comment by fpscan** [comment]
> The problem is, you cannot use Nightly builds, you would have to change the hash every day...

Thank you for your effort. Ah, in this case we are going to ninja release on stable build so we can use stable version in here instead nightly. Is there any roadmap that changes how hash handles for everyday change in future?

**Review by ItzLevvie — APPROVED** [review]

**Pull request closed without merging**